src/librustc_resolve/late.rs

```diff
@@ -374,7 +377,18 @@ impl<'a, 'tcx> Visitor<'tcx> for LateResolutionVisitor<'a, '_> {
         self.resolve_expr(expr, None);
     }
     fn visit_local(&mut self, local: &'tcx Local) {
+        debug!("visit_local {:?} {:?} {:?}", local, local.pat, local.pat.kind);
+        let val = match local {
+            Local { pat, ty: Some(ty), init: None, .. } => match pat.kind {
+                // We check for this to avoid tuple struct fields.
+                PatKind::Wild => None,
```

Comment:
We have `let _: <type>` here -- what's special about that?

Reply:
I've seen no way to differentiate between `let _: <type>;` and `struct T(usize);`. Without this filter the suggestion triggers incorrectly for the latter in one of the tests.